### PR TITLE
BFD-3158: Duplicate New SSM Parameters

### PR DESCRIPTION
**Pull request by brandoncruz3**
**JIRA Ticket:**
[BFD-3158](https://jira.cms.gov/browse/BFD-3158)

**User Story or Bug Summary:**

Background

first step is introducing the new paths as primary paths, and maintaining the existing paths as legacy paths. By primary, I mean the hierarchies that include the literal, anchored definition that is copied to the legacy path via dereference, e.g. for the notional `primary.path.key: value` to the notional 
```
primary:
  path:
    key: &primaryPathKey value
legacy:
  path:
    key: *primaryPathKey 
```

We think this is a generally good idea because:

* we get to evaluate the solution without much effort–if the anchoring/dereferencing is inaccurate, it will fail
* it's an atomic, shippable unit that can continue to be used by all of the SSM hierarchy lookups we already have in place
* it's a significantly smaller change set, largely limited to the base terraservice, rather than... all of the terraservices we currently have defined at once
* subsequent PRs can limit the changes to individual terraservices if we think that's necessary, or do all of the lookups in a single change set, depending on how awful that diff is
* a final PR could be limited to dropping the legacy paths... with or without removing the &anchors. Either way, that change should be really easy to evaluate for the reviewers

AC

Duplicate New SSM Parameters by implementing literal, anchoring definition from the new primary path to the legacy path

---

### What Does This PR Do?

This PR create the new SSM Parameters based on the [Application Variable Inventory wiki ](https://github.com/CMSgov/beneficiary-fhir-data/wiki/Application-Variable-Inventory) doc. We found that anchoring secure values does not work with our current configuration so simply duplicating the values on the new parameters was the route here. 

### What Should Reviewers Watch For?

If you're reviewing this PR, please check for these things in particular:

* Verify all PR security questions and checklists have been completed and addressed.
* It looks like we create around 91 total new parameters per environment. 

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist

I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: ops/terraform/services/base/values/test.yaml [hunk 1]
```diff
@@ -156,13 +249,60 @@ bfd:
...
+      paths:
+        dirs:
+          logs: /usr/local/bfd-server/
+          tmp: /usr/local/bfd-server/tmp
+        files:
+          keystore: /usr/local/bfd-server/bluebutton-appserver-keystore.pfx
+          truststore: /usr/local/bfd-server/bluebutton-appserver-truststore.pfx
+          war: /usr/local/bfd-server/basename('bfd-server-war-1.0.0-SNAPSHOT.war')
+          launcher_script: "bfd-server-launcher-1.0.0-SNAPSHOT/bfd-server-launcher.sh"
```
Comment (mjburling) [inline ops/terraform/services/base/values/test.yaml:295]:
```suggestion
```

File: ops/terraform/services/base/values/test.yaml [hunk 2]
```diff
@@ -58,6 +58,34 @@ bfd:
...
+          license_key: UNDEFINED
+          host: UNDEFINED
+          path: UNDEFINED
+          period: UNDEFINED
+      paths:
+        dirs:
+          logs: /opt/bfd-db-migrator
+          tmp: /opt/bfd-db-migrator/tmp
+        files:
+          launcher_script: /opt/bfd-db-migrator/bfd-db-migrator-1.0.0-SNAPSHOT/bfd-db-migrator.sh
```
Comment (mjburling) [inline ops/terraform/services/base/values/test.yaml:81]:
```suggestion
```

File: ops/terraform/services/base/values/test.yaml [hunk 3]
```diff
@@ -90,7 +118,52 @@ bfd:
...
         data_pipeline_rda_grpc_port: <<CIPHER>>AgV4MTfMbGC8BEsdh-tvW_SwNWWflHfgYsWkH_YayYaISEYAdAACAAtBcHBsaWNhdGlvbgAGY2lwaGVyABVhd3MtY3J5cHRvLXB1YmxpYy1rZXkAREF5QjI3NVpjaWpVaWdsZnJHb2NyOXlhVko5YnlpNi9ydDg1MDRzM2JsU3ZQV2ZxSWlKdEtsVUlyUEJndmQrc3Fxdz09AAEAB2F3cy1rbXMAS2Fybjphd3M6a21zOnVzLWVhc3QtMTo1NzczNzM4MzE3MTE6a2V5L21yay0xNWZhNGU2MzcxOTI0YmZmYTk3NjRjMjE4ZjdhNzY3YQC4AQIBAHic22-xSw25t2N5FjIsl0BfswHAJC80QDtF4qY0HagMgAEEN3CwefKsw7iBHPmlVAeGAAAAfjB8BgkqhkiG9w0BBwagbzBtAgEAMGgGCSqGSIb3DQEHATAeBglghkgBZQMEAS4wEQQMFofLnwGRVtsiXxqVAgEQgDsYSenLSoBfY_JUSXv28mY-Xo3P3CXUggM7SwrlXHHdX-BTwXaZdQgPLjEZ5Rjqd2igrfeSun_r5R0wyQIAABAAG-umRLmghhPxtDyrtkg7IHX_o8EpVmVUWI7gqui13IcBm2YcU-wx53yOhmlL4_iR_____wAAAAEAAAAAAAAAAAAAAAEAAAAFSZ3a1MTrzrA6zypyiF-MZy9e619jAGcwZQIwN6tgMSkdHF_Gqvj48v16Rdx8Wl_d9pQKYYWmekBxUaSi4gKCtKuXck6i_fB5-OnRAjEA3WfEFgat2mtIz-ZnCwst_uOFQs_oohKOMEvpYVrmB-pbRe7GFtZAnpkkvhkMraQK<</CIPHER>>
+      env_name: *env_name
+      paths:
+        files:
+          launcher_script: /bluebutton-data-pipeline/bfd-pipeline-app-1.0.0-SNAPSHOT/bfd-pipeline-app.sh
```
Comment (mjburling) [inline ops/terraform/services/base/values/test.yaml:124]:
```suggestion
```

**Review by mjburling — CHANGES_REQUESTED** [review]
First pass. Drop those hard-coded bits and make an update to the variable inventory.

**Review by mjburling — CHANGES_REQUESTED** [review]
On abandoning the suggested anchoring strategy, this should still be feasible. As per the example I provided, note that the anchor's definition (`&primaryPathKey`) is written *before* aliasing/usage (`*primaryPathKey`). I believe the yaml specification allows for both anchor definition **and** redefinition, so each alias references only the most recent anchor. If it doesn't yet exist, the yaml cannot be parsed.

Can you try adjusting the order of operations before we abandon it?

**Comment by brandoncruz3** [comment]
> On abandoning the suggested anchoring strategy, this should still be feasible. As per the example I provided, note that the anchor's definition (`&primaryPathKey`) is written _before_ aliasing/usage (`*primaryPathKey`). I believe the yaml specification allows for both anchor definition **and** redefinition, so each alias references only the most recent anchor. If it doesn't yet exist, the yaml cannot be parsed.
> 
> Can you try adjusting the order of operations before we abandon it?

💥 You figured it out. I will get these in the right order and take it slow since the bad output doesn't tell me where its barking. Ill update this shortly.

**Review by keithdadkins — APPROVED** [review]
I decrypted, sorted, and diffed all three primary environments and verified that this pr only adds values. No updates or deletes are being made to prod, prod-sbx, or test.yaml

**Review by billie-rose — APPROVED** [review]
Verified locally that only new values were being added during a `terraform plan` in `test`, `prod-sbx`, and `prod`

**Pull request merged**